### PR TITLE
GEODE-7312: modify the ThreadsMonitor to print the stack of a blocking…

### PR DESCRIPTION
**Pull request by bschuchardt**
… thread

Log a thread trace for a thread that's blocking a "stuck" thread.

This will help a lot when a user experiences hung operations.  Prior to
this change we needed to request thread dumps for servers and that was
usually not possible to obtain because the servers had already been
terminated & restarted.  This change puts the relevant thread dumps in
the server's log file, which is much easier for folks to gather after
the fact.

@yossireg 

Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [x] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [x] Is your initial contribution a single, squashed commit?

- [x] Does `gradlew build` run cleanly?

- [x] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, check Concourse for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

File: geode-core/src/main/java/org/apache/geode/internal/monitoring/executor/AbstractExecutor.java [hunk 1]
```diff
@@ -47,53 +48,68 @@ public AbstractExecutor(ThreadsMonitoring tMonitoring, long threadID) {
 
   public void handleExpiry(long stuckTime) {
     this.incNumIterationsStuck();
-    logger.warn(handleLogMessage(stuckTime));
+    logger.warn(createThreadReport(stuckTime));
   }
 
-  private String handleLogMessage(long stuckTime) {
+  String createThreadReport(long stuckTime) {
```
Comment (echobravopapa) [inline geode-core/src/main/java/org/apache/geode/internal/monitoring/executor/AbstractExecutor.java]: like the renames

File: geode-core/src/main/java/org/apache/geode/internal/monitoring/executor/AbstractExecutor.java [hunk 2]
```diff
@@ -47,53 +48,68 @@ public AbstractExecutor(ThreadsMonitoring tMonitoring, long threadID) {
...
+      if (lockOwnerThread != null) {
+        writeThreadStack(lockOwnerThread, LOCK_OWNER_THREAD_STACK, stringBuilder);
       }
     }
 
-    return strb.toString();
+    return stringBuilder.toString();
+  }
+
+  private void writeThreadStack(ThreadInfo thread, String header, StringBuilder strb) {
```
Comment (echobravopapa) [inline geode-core/src/main/java/org/apache/geode/internal/monitoring/executor/AbstractExecutor.java]: nice method extraction

File: geode-core/src/main/java/org/apache/geode/internal/monitoring/executor/AbstractExecutor.java [hunk 3]
```diff
@@ -47,53 +48,68 @@ public AbstractExecutor(ThreadsMonitoring tMonitoring, long threadID) {
...
 
     if (logThreadDetails) {
-      strb.append("Thread Stack:").append(System.lineSeparator());
-      for (int i = 0; i < thread.getStackTrace().length; i++) {
-        String row = thread.getStackTrace()[i].toString();
-        strb.append(row).append(System.lineSeparator());
+      writeThreadStack(thread, "Thread stack:", stringBuilder);
+    }
+
+    if (logThreadDetails && thread.getLockOwnerName() != null) {
```
Comment (echobravopapa) [inline geode-core/src/main/java/org/apache/geode/internal/monitoring/executor/AbstractExecutor.java]: what is the situation when these could/would be null?

Comment (bschuchardt) [reply geode-core/src/main/java/org/apache/geode/internal/monitoring/executor/AbstractExecutor.java]: If there's no thread blocking the stuck thread there won't be a lockOwnerName

**Pull request closed without merging**